Fixed #629

hideki opened the pull request:
- add unit test for newly added utility function.
- update existing unit test for potential test failure.

pasin commented:
@hideki I don't understand the root cause of the unit test failure so I don't think I can review this. But I saw some codes got commented out so if those are not applicable anymore we should remove it.

hideki commented:
https://github.com/couchbase/couchbase-lite-android/commit/a68e8fe8879f545fd95e573bfd0624d8718cc2df => Although Batcher makes tasks processed in a batch. However, because of machine slowness, timing, or other reasons, requests might be contain only one document.  Deleted line of codes could make test failure.

https://github.com/couchbase/couchbase-lite-android/commit/0bfdabdb08761e93e40de17c4c22d4737cf0a186 => In unit test, testContinuousPushReplicationGoesIdleTwice(), local check point may not be sent immediately. Then, deleted lines of codes could make test takes long time to finish.
